cvat-cli/src/cvat_cli/_internal/agent.py

```diff
@@ -108,6 +112,73 @@ class _NewReconnectionDelay:
     delay: timedelta
 
 
+class _TaskCacheLimiter:
+    """
+    This class deletes least-recently used tasks from the dataset cache,
+    so that at any time the cache contains at most _MAX_TASKS_WITH_CHUNKS
+    tasks with downloaded chunks, and at most _MAX_TASKS_WITHOUT_CHUNKS without.
+
+    This helps manage disk usage, since agents may run indefinitely, and
+    we don't want the dataset cache to keep growing.
+    """
+
+    _MAX_TASKS_WITH_CHUNKS = 1
+    _MAX_TASKS_WITHOUT_CHUNKS = 10
```

Comment:
If we're trying to manage disk space, why not limit the maximum amount of memory occupied? If we limit the number or tasks, we don't provide memory guarantees. We have implicit server-side restrictions on chunk size (Redis' 512MB), so if we limit the storage used by, e.g., 1 GB by default, we will have limited cache space, yet we'll always be able to serve a request (by cleaning the cache until it can fit the requested data). I'd always try to download the missing requested chunk, even though it can be bigger than the maximum cache size.

Reply:
I was actually thinking about adding another media policy, where the dataset would only keep one chunk downloaded at a time. (Or maybe two, one current and one prefetched.) That would solve the media storage problem and simplify the cache cleanup code. I don't think caching all chunks is really that useful for an agent, because the entire cache is invalidated when a task's `updated_at` increases.

(However, that only concerns batch requests, so I left it out of this PR.)

Reply:
We can already use `/api/tasks|jobs/{id}/meta .chunks_updated_date` for finer cache management.

Reply:
Hmm, interesting. That could be quite useful for the dataset API.

However, I still would like to move away from caching chunks on the agent in the future. If we want to stay within a certain storage limit (and I think we do), then we won't be able to store all chunks for a task if it's big enough. And then we'd kinda have to fetch chunks on-demand anyway. I suppose it's possible to cache _some_ chunks within a given limit and load others on-demand, but it feels too complicated.